### PR TITLE
S114 jose di for discussion

### PR DESCRIPTION
**Pull request by eschultink**
created just for discussion purposes, as to how to incorporate DI into main branch with all the aws stuff

### CODE REVIEW

File: java/core/src/main/java/co/worklytics/psoxy/HashUtils.java [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 
+import lombok.NoArgsConstructor;
 import org.apache.commons.codec.digest.DigestUtils;
 import org.apache.commons.lang3.StringUtils;
 
+import javax.inject.Inject;
 import java.nio.charset.StandardCharsets;
 import java.util.Base64;
 
+@NoArgsConstructor(onConstructor = @__({@Inject}))
```
Comment (eschultink) [inline java/core/src/main/java/co/worklytics/psoxy/HashUtils.java:11]: why not `@NoArgsConstructor(onConstructor_ = @Inject)` ? cleaner imho

File: java/core/src/main/java/co/worklytics/psoxy/SanitizerFactory.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+import co.worklytics.psoxy.impl.SanitizerImpl;
+import dagger.assisted.AssistedFactory;
+
+@AssistedFactory
+public interface SanitizerFactory {
+
+    SanitizerImpl create(Sanitizer.Options options);
+
+}
```
Comment (eschultink) [inline java/core/src/main/java/co/worklytics/psoxy/SanitizerFactory.java:11]: 👍  yes, I'm porting this

File: java/core/src/main/java/co/worklytics/psoxy/di/PsoxyModule.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package co.worklytics.psoxy.di;
```
Comment (eschultink) [inline java/core/src/main/java/co/worklytics/psoxy/di/PsoxyModule.java:1]: I find naming package `di` like this is rather weird ... why not just put it at top-level?

File: java/core/src/main/java/co/worklytics/psoxy/impl/SanitizerImpl.java [hunk 4]
```diff
@@ -33,9 +35,18 @@
 import java.util.stream.Collectors;
 
 @Log
-@RequiredArgsConstructor
 public class SanitizerImpl implements Sanitizer {
 
+    @AssistedInject
+    public SanitizerImpl(HashUtils hashUtils, @Assisted Options options) {
+        this.options = options;
```
Comment (eschultink) [inline java/core/src/main/java/co/worklytics/psoxy/impl/SanitizerImpl.java:42]: why doesn't this assign `hashUtils`? does that happen magically in `AssistedInject`?

File: java/core/src/main/java/co/worklytics/psoxy/impl/SanitizerImpl.java [hunk 5]
```diff
@@ -107,9 +108,6 @@ public String sanitize(@NonNull URL url, @NonNull String jsonResponse) {
             compiledPseudonymizationsWithOriginals =
                 compile(options.getRules().getPseudonymizationWithOriginals());
         }
```
Comment (eschultink) [inline java/core/src/main/java/co/worklytics/psoxy/impl/SanitizerImpl.java:110]: blocks above are initialization, which should move into DI factory??

File: java/core/src/main/java/co/worklytics/psoxy/HashUtils.java (hunk 1 above)
Comment (jlorper) [reply java/core/src/main/java/co/worklytics/psoxy/HashUtils.java:11]: yeah, is confusing that Lombok's doc still points to `@__({@Inject}))`
> To put annotations on the generated constructor, you can use onConstructor=@__({@AnnotationsHere}), but be careful; this is an experimental feature. For more details see the documentation on the onX feature.

But yes, that's JDK7 style, JDK8 way cleaner. I always get confused about what's what

File: java/core/src/main/java/co/worklytics/psoxy/di/PsoxyModule.java (hunk 3 above)
Comment (jlorper) [reply java/core/src/main/java/co/worklytics/psoxy/di/PsoxyModule.java:1]: I prefer not to mix it with the "real" code. DI is just glue for the app, you can really avoid the package while scanning contents, so I prefer to put it in their on package. But whatever.

File: java/core/src/main/java/co/worklytics/psoxy/impl/SanitizerImpl.java (hunk 4 above)
Comment (jlorper) [reply java/core/src/main/java/co/worklytics/psoxy/impl/SanitizerImpl.java:42]: yes. With `Assisted` you just tell the factory the dependencies that the caller must satisfy. Under the hood this is just like passing "options" to a `@Provides` method.

File: java/core/src/main/java/co/worklytics/psoxy/impl/SanitizerImpl.java (hunk 5 above)
Comment (jlorper) [reply java/core/src/main/java/co/worklytics/psoxy/impl/SanitizerImpl.java:110]: No, should be moved to constructor. I'll do it.

**Pull request closed without merging**